### PR TITLE
Move forms to a separate application

### LINKED ISSUE #816: Move user-facing forms to separate URL endpoint amy.swc.org/forms
This is to simplify URLs we can use in the administrative part of AMY.

User-facing forms: forms that don't require people to log in, like workshop submission, profile update request, etc.

### PR DESCRIPTION
**Pull request by pbanaszkiewicz**
This fixes #816 by moving the external forms (ie. forms that can be accessed by non-authorized users) to a separate Django app `extforms`.

Additionally the forms are being moved to a new URL subtree `/forms/` (they used to live under `/workshops/`). Redirections from the old addresses are provided.

### CODE REVIEW

**Comment by chrismedrela** [comment]
I'll review tomorrow.

**Comment by chrismedrela** [comment]
But one question now: could we make all URLs flat, so that there are no `workshops`, `forms` etc. subtrees? That makes URL shorter and we avoid bikeshedding about URL tree structure. :-)

**Comment by pbanaszkiewicz** [comment]
>But one question now: could we make all URLs flat, so that there are no workshops, forms etc. subtrees?

Maybe one day, but what's the point?

File: extforms/views.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
+from django.conf import settings
+from django.core.urlresolvers import reverse, reverse_lazy
+from django.shortcuts import render
+from django.template.loader import get_template
+from django.views.generic import TemplateView
+
+from workshops.forms import (
+	SWCEventRequestForm,
```
Comment (pbanaszkiewicz) [inline extforms/views.py]: Good catch! I was in the middle of configuring my editor so I ended up with mixed tabs and spaces.

**Comment by chrismedrela** [comment]
@pbanaszkiewicz do you want me to review your code? Unfortunately, I won't manage to do this today, but I can be the very first think I'll do on Sunday.

**Comment by pbanaszkiewicz** [comment]
@chrismedrela yes please, tomorrow is fine, and thank you

File: workshops/urls.py [hunk 2]
```diff
@@ -196,4 +187,36 @@
...
+    url(r'^dc/request_selforganized/$',
+        RedirectView.as_view(pattern_name='dc_workshop_selforganized_request', permanent=True),
+        name='old_dc_workshop_selforganized_request'),
+    url(r'^dc/request_selforganized/confirm/$',
+        RedirectView.as_view(pattern_name='dc_workshop_selforganized_request_confirm', permanent=True),
+        name='old_dc_workshop_selforganized_request_confirm'),
+    url(r'^submit/$',
+        RedirectView.as_view(pattern_name='event_submit', permanent=True),
+        name='old_event_submit'),
+    # url(r'^submit/confirm/$',
```
Comment (chrismedrela) [inline workshops/urls.py]: Please add a comment explaining that this piece of code is intentionally commented out.

File: amy/urls.py [hunk 3]
```diff
@@ -44,6 +44,8 @@
 
     # Login with GitHub credentials
     url('', include('social.apps.django_app.urls', namespace='social')),
+
+    url(r'^forms/', include('extforms.urls')),
```
Comment (chrismedrela) [inline amy/urls.py]: Could you move this before GitHub login, so that we go from more specific URL patterns (`'^forms/'`) to more general ones (`''`).

**Comment by pbanaszkiewicz** [comment]
@chrismedrela changes made - please approve.

**Pull request merged**